walletconnectv2/src/main/kotlin/com/walletconnect/walletconnectv2/engine/domain/EngineInteractor.kt

```diff
@@ -513,30 +462,39 @@ internal class EngineInteractor(
         crypto.removeKeys(request.topic.value)
         sequenceStorageRepository.deleteSession(request.topic)
         relayer.unsubscribe(request.topic)
+
         scope.launch { _sequenceEvent.emit(params.toEngineDoDeleteSession(request.topic)) }
     }
 
     private fun onSessionRequest(request: WCRequestVO, params: SessionParamsVO.SessionRequestParams) {
+        Validator.validateSessionRequest(params.toEngineDORequest(request.topic)) { errorMessage ->
+            relayer.respondWithError(request, PeerError.InvalidSessionRequest(errorMessage))
+            return@validateSessionRequest
```

Comment:
```suggestion
            return
```
And make `Validator.validateSessionRequest` inline